lib/src/metta/interpreter.rs

```diff
@@ -216,6 +224,18 @@ impl InterpreterCache {
     }
 }
 
+fn atom_into_vars_mut(atom: &mut Atom) -> impl Iterator<Item=&mut VariableAtom> {
```

Comment:
Should we make these two functions part of atom?  I see nearly identical code is user here:  https://github.com/trueagi-io/hyperon-experimental/blob/6f40ca298f44db751b8f45158c930d3655437609/lib/src/space/grounding.rs#LL298C1-L301C1

Reply:
Yes, I also thought about this. The question is what API would be the best. For example adding additional iterator on vars to the `Atom` doesn't seem right to me. The `atom_into_vars_mut` function is an attempt to represent a specific `filter_map` for the vars on iterator. What do you think about adding the following into `AtomIter` and `AtomIterMut`:
```rs
    pub fn filter_type<T: TryFrom<&'a Atom>>(self) -> impl Iterator<Item=T> + 'a {
        self.filter_map(|a| <T>::try_from(a).ok())
    }
```
then the code in `GroundingSpace` you mentioned looks like:
```rs
        let query_vars = query.iter().filter_type::<&VariableAtom>().cloned().collect();
```
and we even could remove `.cloned()` after fixing the interface of `Bindings::narrow_vars`:
```rs
        let query_vars = query.iter().filter_type::<&VariableAtom>().collect();
```